### DISCUSSION TITLE
Can <Image /> fallback to <img /> when running next export?

### DISCUSSION THREAD

**Discussion by mfix-stripe** [vercel/next.js#35530] [Ideas]
### Describe the feature you'd like to request

I would like the `Image` component (from `next/image`) to fall back to an `img` when running `next export`. 

The reason being I would like to create a component kit that can be used in all Next.js apps: both ones that use `next export` as well as typical Next.js applications. 

Especially with Next.js principle of progressive enhancement, it feels weird that I can't use the `Image` component when creating regular HTML pages. 

### Describe the solution you'd like

I would like the `Image` component (from `next/image`) to fall back to an `img` when running `next export`, and spit out a warning instead of failing the build.

### Describe alternatives you've considered

Removing `Image` from my component kit/site(s).

**Comment by GabenGar** [comment]
The markup of `Image` is drastically different from `<img>` and also varies between prop values, so it's not drop-in replacement.
If there is a way to detect if the code is built with `next export`, then you can write a HOC which will conditionally return the component or the image element and use it in your code instead. This way you will also have a complete control of the styles and the props in both outcomes. And if there is none, then you can add a public environment variable just for that.

**Reply by mfix-stripe** [reply]
> If there is a way to detect if the code is built with next export, then you can write a HOC which will conditionally return the component or the image element and use it in your code instead. This way you will also have a complete control of the styles and the props in both outcomes. And if there is none, then you can add a public environment variable just for that.

@GabenGar the HOC approach wouldn't work in user-land because the build fails if `next/image` is imported, not even used. Removing `next/image` from the build programmatically isn't feasible in this scenario.

I have thought of the approach of using a signal from Next.js, but Next.js doesn't include one (which still wouldn't work b/c of the above^). The env var approach also doesn't work well as it forces all adopting apps to set a new env var to use a frontend library, which isn't a typical pattern.
